FreeRTOS-Plus/Demo/AWS/Device_Defender_Windows_Simulator/Device_Defender_Demo/report_builder.c

```diff
@@ -323,6 +379,185 @@ static eReportBuilderStatus prvWriteConnectionsArray( char * pcBuffer,
+                                                "\"ip_list\":[" );
+            }
+
+            if( !reportbuilderSNPRINTF_SUCCESS( ulCharactersWritten, ulRemainingBufferLength ) )
+            {
+                eStatus = eReportBuilderBufferTooSmall;
+                break;
+            }
+            else
+            {
+                ulRemainingBufferLength -= ulCharactersWritten;
+                pcCurrentWritePos += ulCharactersWritten;
+            }
+
+            /* Write the metric array elements. */
+            for( i = 0; i < pxCustomMetric->ulLength; i++ )
+            {
+                if( pxCustomMetric->eType == eCustomMetricNumberList )
+                {
+                    ulCharactersWritten = snprintf( pcCurrentWritePos,
+                                                    ulRemainingBufferLength,
+                                                    "%lld,",
+                                                    ( long long ) ( pxCustomMetric->xData.pllNumberList )[ i ] );
+                }
+                else if( pxCustomMetric->eType == eCustomMetricStringList )
+                {
+                    ulCharactersWritten = snprintf( pcCurrentWritePos,
+                                                    ulRemainingBufferLength,
+                                                    "\"%s\",",
+                                                    ( pxCustomMetric->xData.ppcStringList )[ i ] );
+                }
+                else if( pxCustomMetric->eType == eCustomMetricIpList )
+                {
+                    ulCharactersWritten = snprintf( pcCurrentWritePos,
+                                                    ulRemainingBufferLength,
+                                                    "\"%u.%u.%u.%u\",",
+                                                    ( unsigned int ) ( ( pxCustomMetric->xData.pulIpList )[ i ] >> 24 ) & 0xFF,
+                                                    ( unsigned int ) ( ( pxCustomMetric->xData.pulIpList )[ i ] >> 16 ) & 0xFF,
+                                                    ( unsigned int ) ( ( pxCustomMetric->xData.pulIpList )[ i ] >> 8 ) & 0xFF,
+                                                    ( unsigned int ) ( pxCustomMetric->xData.pulIpList )[ i ] & 0xFF );
+                }
+
+                if( !reportbuilderSNPRINTF_SUCCESS( ulCharactersWritten, ulRemainingBufferLength ) )
+                {
+                    eStatus = eReportBuilderBufferTooSmall;
+                    break;
+                }
+                else
+                {
+                    ulRemainingBufferLength -= ulCharactersWritten;
+                    pcCurrentWritePos += ulCharactersWritten;
+                }
+            }
+
+            if( eStatus != eReportBuilderSuccess )
+            {
+                break;
+            }
+
+            /* Discard the last comma. */
+            if( pxCustomMetric->ulLength > 0 )
```

Comment:
Why is the length of the metric queried to determine whether the comma should be deleted? 
Is it possible to have a `CustomMetric_t ` with zero length? Would it be better to have an assert check instead to not accept zero length metrics objects?

Reply:
Yes it should be able to have a zero length metric. The list types could be empty.